src/components/missions/joinButton.js

```diff
@@ -0,0 +1,38 @@
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { Button } from 'react-bootstrap';
+import { joinMission, leaveMission } from '../../redux/missions/missions';
+import style from './missions.css';
+
+const JoinButton = ({ status, id }) => {
+  const dispatch = useDispatch();
+  const fields = {
+    'Not a member': {
+      variant: 'outline-secondary',
+      message: 'Join Mission',
+      click: () => { dispatch(joinMission(id)); },
```

Comment:
- [ ] You should handle this as you did with the rockets, a reserved key should appear with its value set to true or false accordingly to comply with the project requirements, please do so.

The selected mission will have an extra key reserved with its value set to true (As you did for the rockets), example:
``` javascript
  const newState = state.map(mission => {
    if(mission.id !== id) 
        return mission;
    return { ...mission, reserved: true };
});
```

Reply:
Thanks for the feedback, functionality is updated.